vtex/ab-tester
# Feature/testing params interface

#### What is the purpose of this pull request?
Create a testing parameters generic interface, so we can create different testing parameters classes to be used in different AB tests. Currently, only one class is created, for the test using conversion as main metric. Also starts saving test type in vbase and adds an optional parameter to `InitializeAbTest`.

#### What problem is this solving?
This makes way for adding revenue as test metric by allowing the creation of different classes implementing `ITestingParameters`.

#### How should this be manually tested?
2 tests:

Test 1:
- Link this version of the app
- Start an AB test
- Check if everything works (all the tester routes and Colossus periodic update route)
- Unlink the app
- Link a new one from master
- Check if everything still works

Test 2:
- Start an AB test
- Check if everything works (all the tester routes and Colossus periodic update route)
- Link this version of the app
- Check if everything still works
 
#### Types of changes
